video/out/vulkan/context_wayland.c

```diff
@@ -99,6 +101,22 @@ static bool wayland_vk_init(struct ra_ctx *ctx)
     if (!ra_vk_ctx_init(ctx, vk, params, VK_PRESENT_MODE_MAILBOX_KHR))
         goto error;
 
+    // Check if user is already using vk_hdr_layer for color management.
+    const char *hdr_wsi = getenv("ENABLE_HDR_WSI");
+    if (hdr_wsi) {
```

Comment:
No need to check that, the layer will not be available if not set. Avoids some internal knowledge of layer configuration.

Some thoughts about Vulkan suport as a whole. I didn't think much about it, but like mentioned in previous comments, gpu/gpu-next is kind of not compatible with this protocol. vo_gpu generally don't know how to pass hdr metadata, gpu-next relies on hdr vulkan surfaces to be available. Now I'm thinking. To make it work correctly we need to teach `--target-colorspace-hint` to communicate with wayland output code to request metadata. I'm not sure how to do it nicely, didn't think really about it. My point here is to make it work there need to be some more logic in gpu-next to know that there is possible to notify colorspace.

That being said, we can have current status of this PR, but frankly except for dmabuf_wayland it needs more to work correctly and likely this part of code will have to be removed.

Very high level, I'm thinking.
1. vo.c provides "colorspace hint" available flag and possibly what colorspaces are available
2. gpu-next.c checks this and depending on this and internal (vulkan) state and `target-colorspace-hint` decides if it should be used. setting some flag in vo context.
3. gpu-next.c is now aware and sets target_params correctly for hdr output
4. this flag checked if color management should be used

I think it should work well. 

P.S. Note that while current state will enable hdr mode in output display, the metadata wouldn't be correct, so likely we will get reports about it. Currently only dmabuf_wayland works fully correct.

Reply:
>No need to check that, the layer will not be available if not set.

The layer shows up for me in the output as long as I have it installed.

I was under the impression it only did actually did something if you set the right parameters for `vo_gpu_next` (`target-colorspace-hint` and so on) and that the wayland code didn't need to be aware of this. e.g. if vo_gpu_next doesn't find an actual usable surface, the target params would just be bt.1886 and then this protocol does nothing. I'm aware that you can force a target-prim and get an incorrect result due to driver limitations but is I don't think there's really anything we can do about that?

Reply:
Oh it's pretty simple to just not use the protocol for `vo_gpu_next` and (?) `vo_gpu` for now if that's all that ultimately needs to be done. `vo_gpu_next` doesn't need to be aware of this  as it can be completely handled in wayland code.

Reply:
Yes, if that's ok with you we can disable it.